### PR TITLE
[Task] added new `PimcoreBundleAdminSupportInterface`

### LINKED ISSUE #14502: Provide dedicated interface for bundles that provide admin UI JS and CSS
### Improvement description

The following methods from `Pimcore\Extension\Bundle\PimcoreBundleInterface` need to be provided by a dedicated interface which is in the `AdminBundle`, e.g. `PimcoreBundleAdminSupportInterface`. 

```php
    public function getJsPaths(): array;
    public function getCssPaths(): array;
    public function getEditmodeJsPaths(): array;
    public function getEditmodeCssPaths(): array;
```

This needs to be done in `10.6` so that we have a BC layer. 
In `11.x` we then need to remove the 4 methods from `PimcoreBundleInterface`. 

This change makes it then possible to better differentiate between the classic ui and the new ui.

### PR DESCRIPTION
**Pull request by Corepex**
…ods in PimcoreBundleInterface

  
  

## Changes in this pull request  
Resolves #14502

### CODE REVIEW

File: lib/Extension/Bundle/PimcoreBundleInterface.php [hunk 1]
```diff
@@ -65,6 +65,8 @@ public function getAdminIframePath();
      * router.
      *
      * @return string[]|RouteReferenceInterface[]
+     *
+     * @deprecated
```
Comment (jdreesen) [inline lib/Extension/Bundle/PimcoreBundleInterface.php]:
```suggestion
     * @deprecated Will be removed in Pimcore 11. Please implement PimcoreBundleAdminSupportInterface instead.
```

For all of them.

Comment (Corepex) [reply lib/Extension/Bundle/PimcoreBundleInterface.php]: done

**Comment by lukmzig** [comment]
Hi @Corepex, I think we should also adapt the documentation [here]( https://github.com/pimcore/pimcore/tree/10.6/doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/05_Pimcore_Bundles), since we are deprecating these methods. What do you think?

**Comment by Corepex** [comment]
> Hi @Corepex, I think we should also adapt the documentation [here](https://github.com/pimcore/pimcore/tree/10.6/doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/05_Pimcore_Bundles), since we are deprecating these methods. What do you think?

good point  - I will add it 👍

**Comment by brusch** [comment]
@Corepex we also need to consider `\Pimcore\Event\BundleManagerEvents`

**Pull request merged**